geopandas/vectorized.pyx

```diff
@@ -1210,6 +1210,26 @@ class GeometryArray(object):
         result.data[idx == -1] = 0
         return result
 
+    def fill(self, idx, value):
+        """ Fill index locations with value
+
+        Value should be a BaseGeometry
+
+        Returns a copy
+        """
+        base = [self]
+        if isinstance(value, BaseGeometry):
+            base.append(value)
+            value = value.__geom__
+        if value is None:
```

Comment:
if -> elif?

(and should we after this raise a ValueError that the passed value is not of correct type ?)